### ISSUE TITLE
Asynchronous components and their chunks are loaded from wrong path (on production)

### ISSUE THREAD

**Issue by flmuel** [roots/sage#3031]
### Terms

- [X] I have read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
- [X] This request is not a duplicate of an existing issue
- [X] I have read the [docs](https://roots.io/docs/) and followed them (if applicable)
- [X] I have seached the [Roots Discourse](https://discourse.roots.io/) for answers and followed them (if applicable)
- [X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

### Description

As I described in issue #2940, I am having issues loading my asynchronous exported components that are registered through React Habitat. These chunks are loading from wrong path. However, this problem only occurs on production.
So I created a repro, which can be found here: https://github.com/flmuel/sage-async-exports-publicPath-bug

### Steps To Reproduce

1. Clone https://github.com/flmuel/sage-async-exports-publicPath-bug
2. Install theme
3. Compare frontend of dev and production

### Expected Behavior

Chunks should load from correct publicPath.
Like on dev environment:
<img width="2048" alt="Screenshot 2022-04-15 at 13 01 06" src="https://user-images.githubusercontent.com/9769279/163566017-20916a64-f7f5-4b78-acb2-820036dba39a.png">

### Actual Behavior

Chunks are loaded from the current URL path. They should be loaded from the public folder.
<img width="2049" alt="Screenshot 2022-04-15 at 13 00 44" src="https://user-images.githubusercontent.com/9769279/163565865-a1a237df-9e00-4590-a341-56843d462373.png">

### Versions

10.1.6

**Comment by jackkemmish** [comment]
I'm getting a similar issue when conditionally importing my JS. 

No imported JS is loading, but I can see the source is being loaded from the current URL too.

I opened a ticket on Roots Discourse but had no response there yet so will keep an eye on this!

[My ticket on discource about the similar issue](https://discourse.roots.io/t/loading-chunk-error-when-importing-conditional-js-files-to-app-js-yarn-build/22938)

**Comment by jackkemmish** [comment]
Someone on the Discourse gave me an answer to use `.setPublicPath('/wp-content/themes/<theme-name>/public/')` in the bud config. It is working now 🥳 but I feel this might be a hack at the moment and shouldn't be needed.

**Comment by flmuel** [comment]
> I am doing this in my package.json instead `"build": "bud build --clean --publicPath '/wp-content/themes/<theme-name>/public/'",`

I know the function, have you in the above linked ticket also already mentioned and had tried (in the bud config for the production build set). Apart from the fact that the function should no longer be used, it also does not lead to a successful result for me, since the assets which have previously loaded correctly contain the public path 2x in the URL and therefore do not load. I would be surprised if it works for you, I had already put many hours in the sand...

**Comment by jackkemmish** [comment]
Yeah just noticed the double up on path for me too 🤦 

Feel this should really work out the box rather than fidling around 😢

**Comment by jackkemmish** [comment]
Hi @flmuel,

I managed to get mine working by using the following:
```
.entry({
  app: {
    import: ['@scripts/app', '@styles/app'],
    publicPath: '/wp-content/themes/undo/public/'
  },
  editor: ['@scripts/editor', '@styles/editor'],
})
```

This works for both dev and prod 👍

**Comment by flmuel** [comment]
@jackkemmish Your solution did not quite work for me, I had to differ whether the build is for production or not

```js
 .entry({
      app: {
        import: ['@scripts/app', '@styles/app'],
        publicPath: app.isProduction
          ? '/wp-content/themes/foo/public/'
          : '/',
      },
      editor: {
        import: [
          '@scripts/editor',
          '@styles/editor'
        ],
        publicPath: app.isProduction
          ? '/wp-content/themes/foo/public/'
          : '/',
      },
      global: {
        import: ['@scripts/global'],
        publicPath: app.isProduction
          ? '/wp-content/themes/foo/public/'
          : '/',
      },
    })
```

**Comment by retlehs** [comment]
should be resolved in bud 5.8.x
